src/components/dapps/ModalCreateDapps.vue

```diff
@@ -284,32 +306,371 @@
     };
 
+    const openOption = ref(false);
+
+    const unit_d = 3;
+    const decimal = 12;
+
+    const formData = reactive<FormData>({
+      endowment: bnToBn(plasmUtils.reduceDenomToBalance(27, unit_d, decimal)),
+      weight: new BN(200000000000),
+      projectName: '',
+      projectDesc: '',
+      webpage: '',
+    });
+
+    const selAccount = ref(0);
+    const toAccount = ref(props.allAccounts[0] as string);
+    const toAddress = ref(props.allAccounts[0] as string);
+    const toAccountName = ref(props.allAccountNames[0]);
+
+    watch(
+      selAccount,
+      () => {
+        toAccount.value = props.allAccounts[selAccount.value] as string;
+        toAccountName.value = props.allAccountNames[selAccount.value];
+        toAddress.value = props.allAccounts[selAccount.value] as string;
+
+        openOption.value = false;
+      },
+      { immediate: true }
+    );
+
+    const shortenToAddress = computed(() => {
+      const address = toAccount.value as string;
+      return `${address.slice(0, 6)}${'.'.repeat(6)}${address.slice(-6)}`;
+    });
+
+    //abi
+    const store = useStore();
+    const api = computed(() => store.getters.api);
+
+    const { abi, onChangeAbi, onRemoveAbi } = useAbi();
+
+    const {
+      fileRef: wasmFromFile,
+      setFile: setWasmFile,
+      validate: isWasmFromFileValid,
+    } = useFile({
+      onChange: onChangeAbi,
+      onRemove: onRemoveAbi,
+      validate: (file) =>
+        file?.data.subarray(0, 4).toString() === '0,97,115,109',
+    });
+
+    const wasm = ref();
+    const isWasmValid = ref(false);
+
+    const onDropFile = (fileState: FileState) => {
+      setWasmFile(fileState);
+    };
+
+    watch(abi, () => {
+      console.log('abc', abi.value);
+      console.log('wasm', abi.value?.project.source.wasm);
+      console.log('constructor', abi.value?.constructors);
+      if (abi.value && isWasm(abi.value.project.source.wasm)) {
+        wasm.value = abi.value.project.source.wasm;
+        isWasmValid.value = true;
+
+        // if (currentName.current === '') {
+        //   setName(`${abi.value.project.contract.name.toString()}.contract`);
+        // }
+        return;
+      }
+
+      if (wasmFromFile.value && isWasmFromFileValid) {
+        wasm.value = compactAddLength(wasmFromFile.value.data);
+        isWasmValid.value = true;
+
+        return;
+      }
+
+      wasm.value = null;
+      isWasmValid.value = false;
+    });
+
+    const upload = async () => {
+      if (
+        formData.projectName === '' ||
+        toAddress.value === '' ||
+        wasm.value === null
+      ) {
+        store.dispatch(ActionTypes.SHOW_ALERT_MSG, {
+          msg: `Please check fields again`,
+          alertType: 'error',
+        });
+        return;
+      }
+
+      const abiData = abi.value as Abi | AnyJson;
+
+      console.log('s', abiData);
+      console.log('w', wasm.value);
+
+      const code = new CodePromise(api?.value, abiData, wasm.value);
```

Comment:
I already tried this, but it seems `createBlueprint` method is deprecated. so, that's why I should had changed my mind to follow `canvas-ui` or `polkadot-explorer` codebase. they are not using this method anywhere.

Reply:
@sirius651 You're right. Sorry for my misinformation!

The smart contract instantiation on chain looks different.
I can see that from this code https://github.com/paritytech/substrate/blob/6aaa03130da8abbe7c0aa5cf4c3f10557713f9d6/frame/contracts/src/lib.rs#L329-L336
you have to send the contract code, endowment, constructor parameter to deploy it to the chain.

The `createBluePrintTx()` has been simplified and now is part of the constructor tx call of the contract.
https://github.com/polkadot-js/api/blob/92bde9de6f65d06dd9bb593627787f7fb2b390ac/packages/api-contract/src/base/Code.ts#L50

So your method of using `CodePromise.tx.<contractor>.method()` is correct.
https://github.com/polkadot-js/apps/blob/master/packages/page-contracts/src/Codes/Upload.tsx#L90

But I still can't seem to deploy my contracts... I'll try to look into this again!
Having said that, it would be great if you can organize the code structure by separating the contract code-related functions as a `useHook` function or a separate component like this https://github.com/polkadot-js/apps/tree/master/packages/page-contracts/src/Codes. The code is getting too long to debug...